inc/saiport.h

```diff
@@ -501,6 +501,25 @@ typedef enum _sai_port_host_tx_ready_status_t
     SAI_PORT_HOST_TX_READY_STATUS_READY
 } sai_port_host_tx_ready_status_t;
 
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_PATH_TRACING_TIMESTAMP_TYPE
+ */
+typedef enum _sai_port_path_tracing_timestamp_type_t
+{
+    /** Timestamp nanosecond bits [8:15] */
+    SAI_PORT_PATH_TRACING_TIMESTAMP_TYPE_8_15,
+
+    /** Timestamp nanosecond bits [12:19] */
+    SAI_PORT_PATH_TRACING_TIMESTAMP_TYPE_12_19,
+
+    /** Timestamp nanosecond bits [16:23] */
+    SAI_PORT_PATH_TRACING_TIMESTAMP_TYPE_16_23,
+
+    /** Timestamp nanosecond bits [20:27] */
+    SAI_PORT_PATH_TRACING_TIMESTAMP_TYPE_20_27,
+
+} sai_port_path_tracing_timestamp_type_t;
```

Comment:
why those fields are in bits? can't this be a singel 32 i 64 bit integer?

Reply:
@kcudnik 
In Path Tracing [1], midpoint nodes record a truncated time stamp. The Truncated time stamp is 8-bits. 
This enum defines which 8-bits of the timestamp that the midpoint should record. 

[1]https://datatracker.ietf.org/doc/html/draft-filsfils-spring-path-tracing-04

Reply:
so its a hardware limitation or protocol/design limitation ?

Reply:
It’s a protocol design (not a limitation). 
Every PT Midpoint along the packet delivery path records its Midpoint Compressed Data (MCD).
The MCD is designed to be very compact which includes interface ID and Truncated Timestamp (8bits)

The Regional collector (defined in the IETF draft) uses the midpoint truncated timestamp to construct the timestamp of the midpoint.

Reply:
lol seems confusing to me XD